lib/index.js

```diff
@@ -46,55 +44,22 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
  * @param {string} options.keyId - The key ID to use.
  * @param {Object} options.operation - The KMS operation.
  *
- * @returns {Promise<Object>} An object containing `{id}`.
+ * @returns {Promise<Object>} Key information.
  */
 exports.generateKey = async ({keyId, operation}) => {
   assert.string(keyId, 'options.keyId');
   assert.object(operation, 'options.operation');
 
-  let key;
-
   const {invocationTarget: {type}} = operation;
-  if(type === 'AesKeyWrappingKey2019') {
-    // TODO: support other lengths?
-    key = {
-      id: keyId,
-      type,
-      secret: base64url.encode(crypto.randomBytes(32))
-    };
-  } else if(type === 'Sha256HmacKey2019') {
-    // TODO: support other hashes?
-    key = {
-      id: keyId,
-      type,
-      secret: base64url.encode(crypto.randomBytes(32))
-    };
-  } else {
-    throw new Error(`Unknown key type "${type}".`);
-  }
+  assert.string(type, 'options.operation.type');
 
-  // insert the key and get the updated record
-  const now = Date.now();
-  const meta = {created: now, updated: now};
-  const record = {
-    id: database.hash(keyId),
-    meta,
-    key
-  };
-  try {
-    await database.collections.ssm.insert(record, database.writeOptions);
-    return {id: keyId};
-  } catch(e) {
-    if(!database.isDuplicateError(e)) {
-      throw e;
-    }
-    throw new BedrockError(
-      'Duplicate key identifier.',
-      'DuplicateError', {
-        public: true,
-        httpStatusCode: 409
-      }, e);
+  if(_constants.KEY_TYPES.asymmetric.has(type)) {
+    return _asymmetricKey.generateKeyPair({keyId, type});
+  } else if(_constants.KEY_TYPES.symmetric.has(type)) {
```

Comment:
Don't need `else` when returning early. Another pattern to consider would be to map key types to the function to call and that could remove the extra conditional... but that's just a refactoring option, not necessary.

Reply:
Fixed.